kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -57,10 +66,13 @@ private Provisioner() {}
      * @return number of topics created
      * @throws ProvisioningException on provision failure
      */
-    public static int provisionTopics(final KafkaApiSpec apiSpec, final Admin adminClient) {
+    public static TopicProvisionStatus provisionTopics(
+            final KafkaApiSpec apiSpec, final Admin adminClient) {
 
         final var domainTopics = apiSpec.listDomainOwnedTopics();
 
+        final var status = TopicProvisionStatus.builder().domainTopics(domainTopics);
+
```

Comment:
If you refactor each of the chunks of code below into methods, then this method could just be:

```java
public static TopicProvisionStatus provisionTopics(
    final KafkaApiSpec apiSpec, final Admin adminClient) {

  final var existing = existingTopics(adminClient);
  final var created = createTopics(adminClient, apiSpec, existing);

  return TopicProvisionStatus.builder()
     .domainTopics(apiSpec.listDomainOwnedTopics())
     .existingTopics(existing)
     .createTopics(created)
     .build();
}
```

Which would be **much** easier to read!

Reply:
yep - done